### ISSUE TITLE
create_oauth2_access_token() missing?

### ISSUE THREAD

**Issue by doRadiology** [dropbox/dropbox-sdk-obj-c#46]
Will the Objective-c SDK implement the /token_from_oauth1 endpoint to migrate an OAuth 1 tokens to an OAuth 2 tokens?

Thanks

**Comment by scobbe** [comment]
@doRadiology: We're in the process of setting up an API v2 endpoint that is analogous to /token_from_oauth1, in addition to a migration tool in the SDK that should automatically convert your users' stored OAuth 1.0 tokens to OAuth 2.0 tokens.

We don't have an exact timetable for when the endpoint and the tool will be done, but it is our number one priority with the SDK right now, so hopefully we should have it out fairly soon!

**Comment by doRadiology** [comment]
That sounds hopeful, thanks.

**Comment by cesgarma** [comment]
Hi. Is there an update on the progress for this migration tool??
Thanks,
Cesar

**Comment by greg-db** [comment]
The endpoint, [/2/auth/token/from_oauth1](https://www.dropbox.com/developers/documentation/http/documentation#auth-token-from_oauth1), is now available, but it's not implemented in this SDK yet.

**Comment by cesgarma** [comment]
Thanks for the update. I am looking forward to this feature being in the SDK. Any expected time frame for implementation on this SDK?

**Comment by greg-db** [comment]
I don't have a timeline to offer right now.

**Comment by scobbe** [comment]
@cesgarma: This should be available in the latest version of the SDK (2.0.0). Please be sure to setup a custom `DBTransportClient` with your app key and secret. The interface is a little bit clumsy right now. Sorry about that.

```
DBTransportClient *transportClient = [[DBTransportClient alloc] initWithAccessToken:nil
                                                                           selectUser:nil
                                                                            baseHosts:nil
                                                                            userAgent:nil
                                                                  backgroundSessionId:nil
                                                                        delegateQueue:nil
                                                                               appKey:<YOUR DROPBOX APP KEY>
                                                                            appSecret:<YOUR DROPBOX APP SECRET>;

[DropboxClientsManager setupWithAppKey:data.fullDropboxAppKey transportClient:transportClient];
```

Something like this. Let me know if you run into any issues. Thanks for your patience!

**Comment by greg-db** [comment]
Confirmed this is working for me. To elaborate on scobbe's example:

    DBTransportClient *transportClient = [[DBTransportClient alloc] initWithAccessToken:nil
                                                                             selectUser:nil
                                                                              baseHosts:nil
                                                                              userAgent:nil
                                                                    backgroundSessionId:nil
                                                                          delegateQueue:nil    
                                                                          appKey:@"APP_KEY"
                                                                       appSecret:@"APP_SECRET"];

    [DropboxClientsManager setupWithAppKey:@"APP_KEY" transportClient:transportClient];

...

    [[client.authRoutes tokenFromOauth1:@"OAUTH1_ACCESS_TOKEN_KEY" oauth1TokenSecret:@"OAUTH1_ACCESS_TOKEN_SECRET"] response:^(DBAUTHTokenFromOAuth1Result *result, DBAUTHTokenFromOAuth1Error *routeError, DBRequestError *error) {
        if (result) {
            NSLog(@"OAuth 2 access token: %@\n", result.oauth2Token);
        } else {
            NSLog(@"%@\n%@\n", routeError, error);
        }
    }];
